library/spdm_requester_lib/libspdm_req_send_receive.c

```diff
@@ -298,6 +305,60 @@ libspdm_return_t libspdm_receive_spdm_response(libspdm_context_t *spdm_context,
         }
     }
 
-    return libspdm_receive_response(spdm_context, session_id, false,
-                                    response_size, response);
+    #if !LIBSPDM_ENABLE_CHUNK_CAP
+    status = libspdm_receive_response(spdm_context, session_id, false,
+                                      response_size, response);
+    #else /* LIBSPDM_ENABLE_CHUNK_CAP */
+    response_capacity = *response_size;
+    status = libspdm_receive_response(spdm_context, session_id, false,
+                                      response_size, response);
+
+    spdm_response = (spdm_message_header_t*) (*response);
+
+    if (*response_size < sizeof(spdm_message_header_t)) {
+        status = LIBSPDM_STATUS_INVALID_MSG_SIZE;
+        goto receive_done;
+    }
+
+    if (spdm_response->request_response_code == SPDM_ERROR
+        && spdm_response->param1 == SPDM_ERROR_CODE_LARGE_RESPONSE) {
+
+        status = libspdm_handle_error_large_response(
+            spdm_context, NULL,
+            response_size, (void*) spdm_response, response_capacity);
+
+        if (LIBSPDM_STATUS_IS_ERROR(status)) {
+            status = LIBSPDM_STATUS_RECEIVE_FAIL;
+            goto receive_done;
+        }
+
+        if (*response_size < sizeof(spdm_message_header_t)) {
+            status = LIBSPDM_STATUS_INVALID_MSG_SIZE;
+            goto receive_done;
+        }
+
+        /* These are the expected possible large response types.
+         * Anything else is potentially an unexpected error.
+         * If this assert pops, evaluate if the type should be added here
+         * according to the spec, or if it an error by the responder. */
+        LIBSPDM_ASSERT(
```

Comment:
These will eventually need to be part of production code to detect if a Responder attempts to chunk a response that guaranteed to be less than `MinDataTransferSize`. But for now we can keep them as assertions.

Reply:
ok.